src/scheduler/Executor.cpp

```diff
@@ -518,40 +520,19 @@ void Executor::threadPoolThread(int threadPoolIdx)
             }
 
             // Fill snapshot gaps with overwrite regions first
-            std::string mainThreadSnapKey =
-              faabric::util::getMainThreadSnapshotKey(msg);
             auto snap = reg.getSnapshot(mainThreadSnapKey);
             snap->fillGapsWithBytewiseRegions();
 
             // Compare snapshot with all dirty regions for this executor
-            std::vector<faabric::util::SnapshotDiff> diffs;
             {
                 // Do the diffing
                 faabric::util::FullLock lock(threadExecutionMutex);
                 diffs = snap->diffWithDirtyRegions(memView, dirtyRegions);
                 dirtyRegions.clear();
             }
 
-            if (diffs.empty()) {
-                SPDLOG_DEBUG("No diffs for {}", mainThreadSnapKey);
-            } else {
-                // On master we queue the diffs locally directly, on a remote
-                // host we push them back to master
-                if (isMaster) {
-                    SPDLOG_DEBUG(
-                      "Queueing {} diffs for {} to snapshot {} (group {})",
-                      diffs.size(),
-                      faabric::util::funcToString(msg, false),
-                      mainThreadSnapKey,
-                      msg.groupid());
-
-                    snap->queueDiffs(diffs);
-                } else if (isLastInBatch) {
-                    sch.pushSnapshotDiffs(msg, mainThreadSnapKey, diffs);
-                }
-            }
```

Comment:
This was the original place where the snapshot diff pushing was happening, it's now been moved later in the method, and this logic lives in the scheduler now.